sklearn/model_selection/tests/test_validation.py

```diff
@@ -936,6 +937,138 @@ def test_cross_val_predict_with_method():
         assert_array_almost_equal(expected_predictions, predictions)
 
 
+def test_cross_val_predict_class_subset():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X, y = shuffle(X, y, random_state=0)
+    classes = len(set(y))
+
+    # Modifies the dataset so that in a particular fold, produced by
+    # kfold.split, the training set is composed of only 2 classes instead of 3
+    y[:50] = 0
+    y[50:100] = 1
+    X[:100], y[:100] = shuffle(X[:100], y[:100], random_state=0)
```

Comment:
(I've corrected cv=3 to cv=KFold(3))

To be sure, what you've got here looks like it tests the issue, but it's not easy to see that it does so.